api/src/main/kotlin/io/spine/protodata/CompilationErrors.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2025, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.protodata
+
+import io.spine.environment.Tests
+import io.spine.protodata.Compilation.ERROR_EXIT_CODE
+import io.spine.protodata.ast.File
+import io.spine.protodata.ast.Span
+import io.spine.protodata.ast.toPath
+
+/**
+ * Prints the error diagnostics to [System.err] and terminates the compilation.
+ *
+ * The termination of the compilation in the production mode is done by
+ * exiting the process with [ERROR_EXIT_CODE].
+ *
+ * If the code is run [under tests][Tests] the method throws [Compilation.Error].
+ *
+ * @param file The file in which the error occurred.
+ * @param span The location span of the erroneous Protobuf declaration.
+ * @param message The description of what went wrong.
+ *
+ * @throws Compilation.Error exception when called under tests.
+ */
+public fun compilationError(file: File, span: Span, message: () -> String): Nothing =
```

Comment:
Can we have it as another function under the `Compilation` object, please?

Reply:
Or, if we insist on having such things as top level functions, rather than mathods of the `Compilation` object, this should be discussed first.

Reply:
The reason we have the `Compilation` object is potential need to cache something internal things. Or, do not terminate immediately on the first error and count them until terminating the compilation.

Such things are more natural to do under a class or an object.